cnf-certification-test/lifecycle/suite.go

```diff
@@ -584,23 +581,35 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 func testStorageRequiredPods(env *provider.TestEnvironment) {
-	oc := clientsholder.GetClientsHolder()
-	var nonCompliantPods []*provider.Pod
-	for _, pod := range env.Pods {
-		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
-			if podOwner.Kind != statefulSet {
-				continue
-			}
-			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
-			if err != nil {
-				tnf.ClaimFilePrintf(err.Error())
+	var podsWithLocalStorage []string
+	for _, put := range env.Pods {
+		localStorageFound := false
+		for pvIndex := range put.Spec.Volumes {
+			// Skip any nil persistentClaims.
+			if put.Spec.Volumes[pvIndex].PersistentVolumeClaim == nil {
 				continue
 			}
-			if statefulset.Spec.ServiceName == localStorage {
-				nonCompliantPods = append(nonCompliantPods, pod)
-				continue
+			// We have the list of pods/volumes/claims.
+			// Look through the storageClass list for a match.
+
+			var StorageClasses = env.StorageClasses
+			var Pvc = env.PersistentVolumeClaims
+			for i := range Pvc {
+				if Pvc[i].Name == put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName {
+					for j := range StorageClasses {
+						if StorageClasses[j].Provisioner == "kubernetes.io/no-provisioner" && !localStorageFound {
+							localStorageFound = true
+							tnf.ClaimFilePrintf("%s has been found to use a local storage enabled storageClass", put.String())
+							podsWithLocalStorage = append(podsWithLocalStorage, put.String())
```

Comment:
instead of checking only the first local volume found, find all of them and report that as a result of this test. So in the end we will have a list of bad volumes instead of bad pods. The string could include the namespace/podname/pvc_name/storageclass_name/provisionner_name per non compliant volume

Reply:
I updated but tell me if is correct or need more changes. @edcdavid

Reply:
Looks good now, I would also print the name of the PVC and storage class in this log

Reply:
I see it now, looks good